### DISCUSSION TITLE
Any way to run custom shell commands using poetry?

### DISCUSSION THREAD

**Discussion by AdityaMayukhSom** [python-poetry/poetry#9822] [General] [answered]
I am asking this because I have two scripts one to test and another to eval some code of mine. I am curious that whether I can define some commands with `&` or with environment variables in `pyproject.toml` file and run those using poetry. I do not want to run custom scripts, I just have a long powershell / bash command, which I need to execute under poetry environment. 

The thing I am asking is similar to how we can define scripts under `package.json` in javascript projects and can call them with `npx run <script_name>`. Thanks in advance.  

P.S: #265 discusses about custom pyhon script, but I am asking about shell scripts in general.

**Comment by finswimmer** [comment, marked as answer]
Hey @AdityaMayukhSom,

something like this has been requested before. See #2496

fin swimmer

**Comment by M4RC0Sx** [comment]
I have created a plugin for this: https://github.com/M4RC0Sx/poetry-plugin-exec



> I am asking this because I have two scripts one to test and another to eval some code of mine. I am curious that whether I can define some commands with `&` or with environment variables in `pyproject.toml` file and run those using poetry. I do not want to run custom scripts, I just have a long powershell / bash command, which I need to execute under poetry environment.
> 
> The thing I am asking is similar to how we can define scripts under `package.json` in javascript projects and can call them with `npx run <script_name>`. Thanks in advance.
> 
> P.S: #265 discusses about custom pyhon script, but I am asking about shell scripts in general.
